Userland/Libraries/LibCMake/Token.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023, Sam Atkins <atkinssj@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StringView.h>
+#include <AK/Vector.h>
+
+namespace CMake {
+
+struct Position {
+    size_t line { 0 };
+    size_t column { 0 };
+};
+
+struct VariableReference {
+    StringView value;
+    Position start;
+    Position end;
+};
+
+enum class ControlKeywordType {
+    If,
+    ElseIf,
+    Else,
+    EndIf,
+    ForEach,
+    EndForEach,
+    While,
+    EndWhile,
+    Macro,
+    EndMacro,
+    Function,
+    EndFunction,
```

Comment:
this is missing continue() and break()

https://cmake.org/cmake/help/v3.26/manual/cmake-language.7.html#loops